### PR TITLE
[Draft] Docs: Update layouts documentation for readability

### PR DESCRIPTION
**Pull request by stichbury**
## Description
I've updated the page to improve readability but I think we now need to iterate it as part of the ongoing effort to use it for dashboard generation with OpenAI??

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/docs/pages/user-guides/layouts.md [hunk 1]
```diff
@@ -1,16 +1,13 @@
 # How to use layouts
 
-This guide shows you how to use the [`Layout`][vizro.models.Layout] to arrange charts/components on the screen and customize the grid specifications.
+The [`Page`][vizro.models.Page] model accepts a `layout` argument so that you can supply it with a custom grid. This guide shows how to use [`Layout`][vizro.models.Layout] to arrange charts and components on the screen so you can customize the grid specifications.
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/user-guides/layouts.md:3]: Feel free to rephrase, but I think the order is a bit confusing here. What the user needs to understand:
1. Use layout to arrange charts and components as you like
2. Use the grid argument of the layout to customise the arrangement

```suggestion
The [`Page`][vizro.models.Page] model accepts a `layout` argument that allows for custom arrangement of charts and components on the screen. This guide shows how to customize the grid sepcifications in the [`Layout`][vizro.models.Layout].
```

File: vizro-core/docs/pages/user-guides/layouts.md [hunk 2]
```diff
@@ -48,42 +45,51 @@ If that is your desired layout, you can create your charts/components without pr
...
-- The area spanned by a chart/component in the grid must be rectangular
-- The grid can be arbitrarily large, allowing arbitrarily granular control of the grid:
-
-```python title="Advanced Example"
-grid=[[0, 1, 3, 4],
-      [2, 2, 3, 4]]
+
+- Each chart/component will take the entire space of its grid area (spaces are not enabled).
+- The area spanned by a chart/component in the grid must be rectangular.
+- The grid can be arbitrarily large, allowing arbitrarily granular control of the grid.
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/user-guides/layouts.md:66]: Spaces are enabled now! The users have to provide -1 to create empty spaces 👍 

```suggestion
- Each chart/component will take the entire space of its grid area.
- The area spanned by a chart/component in the grid must be rectangular.
- The grid can be arbitrarily large, allowing arbitrarily granular control of the grid.
- Empty spaces can be added via providing `-1` in the allocated grid space
```

File: vizro-core/docs/pages/user-guides/layouts.md [hunk 3]
```diff
@@ -48,42 +45,51 @@ If that is your desired layout, you can create your charts/components without pr
...
+
+
+## Vertical and horizontal stacking
+As described above, when no `Layout` is specified, components are presented **vertically** as a single-column stack. If you have three components, the default `Layout.grid` will be as follows:
+
+```python title="Default grid"
+grid = [[0], [1], [2]]
+```
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/user-guides/layouts.md:74]: I know this formatting is a bit ugly, but I think it helps to understand the grid layout
```suggestion
```python title="Default grid"
grid = [
             [0], 
             [1],
             [2]
       ]
```
```

File: vizro-core/docs/pages/user-guides/layouts.md [hunk 4]
```diff
@@ -48,42 +45,51 @@ If that is your desired layout, you can create your charts/components without pr
...
+
+Th layout has three equally sized rows, each containing a component spanning the entire width.
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/user-guides/layouts.md:76]:
```suggestion
The layout has three equally sized rows, each containing a component spanning the entire width.
```

File: vizro-core/docs/pages/user-guides/layouts.md [hunk 5]
```diff
@@ -257,37 +266,37 @@ See our section on [when to use `Containers` vs. `Page.layout`](container.md#whe
 
     [GridAdv]: ../../assets/user_guides/layout/grid_advanced.png
 
-## Use custom layout examples
-Below is a table of examples you can take as a reference to create some selected layouts:
+## Custom layout examples
+Here is a reference table of examples layouts:
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/user-guides/layouts.md:270]:
```suggestion
Here is a reference table of example layouts:
```

**Review by huong-li-nguyen — COMMENTED** [review]
Thanks for rewriting this - I think it's a lot clearer now! Have a few suggestions, but feel free to rephrase the suggestions. Otherwise good to go for me 👍

File: vizro-core/docs/pages/user-guides/layouts.md (hunk 2 above)
Comment (stichbury) [reply vizro-core/docs/pages/user-guides/layouts.md:66]: That confused me later on the page, and makes sense now you say they're enabled. I'll delete this or, rather, rephrase it.

**Review by huong-li-nguyen — APPROVED** [review]

**Review by nadijagraca — APPROVED** [review]
Looks good. ⭐

**Pull request merged**